### PR TITLE
Disable IPv6 for C# Ice/adapterDeactivation

### PR DESCRIPTION
**Pull request by pepone**

### CODE REVIEW

File: scripts/tests/Ice/adapterDeactivation.py [hunk 1]
```diff
@@ -3,7 +3,10 @@
...
 
-from Util import TestSuite
+from Util import CSharpMapping, Darwin, TestSuite, platform, Mapping
 
+# Disable IPv6 for macOS until https://github.com/dotnet/runtime/pull/108334 is merged in .NET 8
+# See https://github.com/zeroc-ice/ice/issues/2061
+ipv6 = not (isinstance(Mapping.getByPath(__name__), CSharpMapping) and isinstance(platform, Darwin))
 
-TestSuite(__name__, multihost=False)
+TestSuite(__name__, multihost=False, options={"ipv6": [ipv6]})
```
Comment (bernardnormier) [inline scripts/tests/Ice/adapterDeactivation.py:12]: When we set ipv6 to true, what happens? Do we still run the test with ipv4?

Comment (pepone) [reply scripts/tests/Ice/adapterDeactivation.py:12]: Here IPv6 means the tests supports ipv6, this affects the combinations that are run when you use `--all`. With this the macOS run with `--all` would not include `--ipv6` in the generated combinations.

**Review by externl — APPROVED** [review]

**Comment by externl** [comment]
CI seems to be failing

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -43,7 +41,7 @@ jobs:
             config: "debug"
             build_flags: "OPTIMIZE=no"
             # https://github.com/zeroc-ice/ice/issues/2061
-            test_flags: "--swift-config=debug --rfilter=csharp/Ice/adapterDeactivation"
+            test_flags: "--swift-config=debug"
```
Comment (bernardnormier) [inline .github/workflows/ci.yml:44]: Remove comment above.

**Review by bernardnormier — APPROVED** [review]

**Pull request merged**